services/client/options.go

```diff
@@ -101,6 +113,10 @@ func (o options) Validate() error {
 		return errNoHeartbeatGen
 	}
 
+	if o.ldGen == nil {
```

Comment:
I think that at least for now it follows the constraints we've set, i.e. we require that a heartbeat gen function be set even though some service clients may only care about kv